lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1234 @@
+    """
+    if collection_method == CollectionMethod.KIOSK:
+        return record["nasal_swab_q"]
+    elif collection_method == CollectionMethod.SWAB_AND_SEND:
+        return record["date_on_tube"] or record["kit_reg_date"] or record["back_end_scan_date"]
+    else:
+        return None
+
+def combine_multiple_fields(record: Dict[Any, Any], field_prefix: str, field_suffix: str = "") -> Optional[List]:
+        """
+        Handles the combining of multiple fields asking the same question such
+        as country and state traveled.
+        """
+        regex = rf'^{re.escape(field_prefix)}[0-9]+{re.escape(field_suffix)}$'
+        empty_value = ''
+        answered_fields = list(filter(lambda f: filter_fields(f, record[f], regex, empty_value), record))
+
+        if not answered_fields:
+            return None
+
+        return list(map(lambda x: record[x], answered_fields))
+
+
+
+def create_enrollment_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the enrollment
+    encounter (i.e. encounter of enrollment into the study)
+    """
+
+    # Do not include `core_age_years` because we calculate the age ourselves in the computed questionnaire.
+    integer_questions = [
+        'weight',
+        'height_total',
+        'tier'
+
+    ]
+
+    string_questions = [
+        'study_area',
+        'text_or_email',
+        'text_or_email_attestation',
+        'campus_location',
+        'affiliation',
+        'student_level',
+        'sea_employee_type',
+        'core_house_members',
+        'core_education',
+        'core_income',
+        'wfh_base',
+        'core_housing_type',
+        'core_health_risk',
+        'core_tobacco_use',
+        'sought_care_base',
+        'hospital_where_base',
+        'hospital_ed_base',
+        'prior_test_positive_base',
+        'prior_test_type_base',
+        'prior_test_result_base',
+        'contact_base',
+        'wash_hands_base',
+        'clean_surfaces_base',
+        'hide_cough_base',
+        'mask_base',
+        'distance_base',
+        'novax_reason',
+        'covid_vaccine',
+        'covid_novax_reason',
+        'countries_visited_base',
+        'states_visited_base',
+        'alerts_off',
+        'pronouns',
+        'on_campus_freq',
+        'vaccine_method',
+        'vaccine_where'
+    ]
+
+    date_questions = [
+        'today_consent',
+        'enrollment_date_time',
+        'hospital_arrive_base',
+        'hospital_leave_base',
+        'prior_test_positive_date_base'
+    ]
+
+    boolean_questions = [
+        'attend_uw',
+        'english_speaking',
+        'athlete',
+        'uw_medicine_yesno',
+        'inperson_classes',
+        'uw_job',
+        'uw_greek_member',
+        'live_other_uw',
+        'uw_apt_yesno',
+        'core_pregnant',
+        'core_latinx',
```

Comment:
Thoughts for the future:
In the past we've ingested this as "ethnicity", and I can't currently recall whether we stored that in a FHIR resource as a primary Resource attribute or a QuestionnaireResponse Item

Reply:
In redcap_det_scan it's a boolean withe the name 'ethnicity;